### PR TITLE
Add a few code tours

### LINKED ISSUE #1111: Introduce code tour which explains how to contribute a new scraper
Introduce code tour which explains the steps that have to be taken to contribute a new scraper.

### Resources

- [CodeTour VSCode extension allows you to produce interactive guides, assessments and tutorials](https://techcommunity.microsoft.com/t5/educator-developer-blog/codetour-vscode-extension-allows-you-to-produce-interactive/ba-p/1274297)
- [CodeTour extension](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour)
- [Example repo](https://github.com/anthonychu/azure-functions-deno-template)

Follow-up of #1110

### PR DESCRIPTION
**Pull request by adamconnelly**
I've added a few CodeTours to Promitor:

- An overview of the repo.
- A description of the different C# projects.
- A tour explaining how to add a new Scraper.

Fixes #1111

### CODE REVIEW

**Comment by tomkerkhove** [comment]
So, how do I test this? :D

**Comment by adamconnelly** [comment]
@tomkerkhove let me know what you think of this - it was pretty easy to setup. I think the best thing for you to do is to checkout my branch and view it in vscode using the CodeTours extension (or maybe VS supports it?).

At the moment I've left out descriptions of a few parts of the repo / projects I wasn't sure about, so maybe you can fill in those gaps. I also didn't include adding resource discovery as part of adding a scraper, partly because I haven't actually done it yet so I don't know what I'm talking about, and partly because I figured it might be a good idea to split them, but link from one to the other. It's just that adding a new Scraper is a big enough job in itself without also adding the discovery part at the same time.

I also haven't updated / removed https://github.com/tomkerkhove/promitor/blob/master/adding-a-new-scraper.md yet. Let me know what you want to do about it.

**Comment by adamconnelly** [comment]
> So, how do I test this? :D

haha - I think you should be able to just pull my branch and open vscode. The first time you open it you should get a popup in the bottom right asking if you want to do a tour. If you don't see that / miss that, there's a CodeTours section that's part of the files view:

image

You can just choose a tour and click the play icon that appears when you hover over it / right click and play.

File: .tours/add-a-new-scraper.tour [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+        "start": {
+          "line": 10,
+          "character": 13
+        },
+        "end": {
+          "line": 10,
+          "character": 48
+        }
+      },
+      "description": "### Map Config Properties\n\nAdd a call to `Map()` for each property you need to deserialize. The deserialization code automatically looks for Yaml fields with the same name as your property, but starting with a lowercase letter. For example, the `QueueName` property can be deserialized from this Yaml:\n\n```yaml\nqueueName: 'promitor-queue'`\n```"
```
Comment (tomkerkhove) [inline .tours/add-a-new-scraper.tour:108]:
```suggestion
      "description": "### Map Config Properties\n\nAdd a call to `Map()` for each property you need to deserialize. The deserialization code automatically looks for Yaml fields with the same name as your property, but starting with a lowercase letter. For example, the `QueueName` property can be deserialized from this Yaml:\n\n```yaml\nqueueName: 'promitor-queue'\n```"
```

File: .tours/add-a-new-scraper.tour [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+        "start": {
+          "line": 14,
+          "character": 17
+        },
+        "end": {
+          "line": 14,
+          "character": 63
+        }
+      },
+      "description": "### Using a Child Deserializer\n\nIf you need to deserialize a complex property rather than a simple field, you can tell the deserialization framework to use a child validator to perform the deserialization. You can pass a parameter of type `IDeserializer<TChild>` into your constructor, and the Dependency Injection framework will automatically provide it for you."
```
Comment (tomkerkhove) [inline .tours/add-a-new-scraper.tour:164]: Maybe this step is relevant for adding a new scraper?

Comment (adamconnelly) [reply .tours/add-a-new-scraper.tour:164]: Just checking - do you mean "not relevant"? If so, yeah, I think we could potentially drop some of these steps. Happy to get rid of it - I'll just wait to double check that's what you meant.

**Comment by adamconnelly** [comment]
> Adding a new scraper feels like jumping through a lot of hoops but I think it's ok how it is

@tomkerkhove yeah - I actually don't think it's too bad, although I'm sure we could simplify things a bit more. I think the number of steps in the tour maybe makes it seem worse than it really is, but I just wanted to highlight a few different things that people might want to think about / use, and I also wanted to include stuff like testing and updating the docs / changelog since it's not difficult to do and otherwise if we don't the burden of doing that stuff will just end up falling on you.

I think it would be really interesting to get some feedback from someone actually adding a scraper for the first time and getting them to use the tour. Not sure how easy that is though :smile: That sort of feedback would also help us identify parts that could do with a bit of refactoring to make it a bit simpler.

File: .tours/add-a-new-scraper.tour (hunk 2 above)
Comment (tomkerkhove) [reply .tours/add-a-new-scraper.tour:164]: Yes but it's ok - We can still remove later on!

**Comment by tomkerkhove** [comment]
I'll ask people to use it to learn how it works; adding Codespaces support will make it even easier and more natively as well.

**Pull request merged**